drivers/volume/portworx/portworx.go

```diff
@@ -450,6 +455,25 @@ func (d *portworx) init(sched, nodeDriver, token, storageProvisioner, csiGeneric
 	} else {
 		torpedovolume.StorageProvisioner = provisioners[torpedovolume.DefaultStorageProvisioner]
 	}
+
+	// Update node PxPodRestartCount during init
+	k8s, err := scheduler.Get(sched)
+	if err != nil {
+		return fmt.Errorf("scheduler with name: [%s] not found. Error: [%v]", sched, err)
+	}
+
+	pxPodRestartCountMap, err := k8s.GetPodsRestartCount(namespace, pxLabel)
+	if err != nil {
+		return fmt.Errorf("unable to get portworx pods restart count. Error: [%v]", err)
+	}
+
+	for pod, value := range pxPodRestartCountMap {
+		n, err := node.GetNodeByIP(pod.Status.HostIP)
+		if err != nil {
+			return err
+		}
+		n.PxPodRestartCount = value
```

Comment:
I think we are missing node.UpdateNode.

Reply:
why UpdateNode is needed when you are updating the object structure field ?

Reply:
I think the node.GetNodeByIP actually returns a copy of the node object, rather than a pointer to the nodeRegistry map in the node package. The node.UpdateNode can help update the nodeRegistry map.

Reply:
Yes you are right, good catch . Thanks !!